### PR TITLE
Feature/ilogical scrolling (fixes #2)

### LINKED ISSUE #2: Handle scrolling
Most of the scrolling code is currently commented out since Avalonia doesn't have `IScrollInfo`. Possibly `ILogicalScrollable` could be made to work here.

Most of the code is in `TextView` (I deleted much of the scroll code, may need to copy again from AvalonEdit).

### PR DESCRIPTION
**Pull request by danwalmsley**
Saved you a massive headache there!

Working scrolling!

### CODE REVIEW

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 1]
```diff
@@ -936,7 +972,104 @@ internal void OnTextCopied(TextEventArgs e)
             TextCopied?.Invoke(this, e);
         }
 
+        private int LogicalScrollSize
+        {
+            get
+            {
+                if (TextView?.Document != null)
+                {
+                    return TextView.Document.LineCount + 2;
```
Comment (aelij) [inline src/AvaloniaEdit/Editing/TextArea.cs]: `2` should probably be a constant (e.g. `MinimumLineCount`)

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 2]
```diff
@@ -936,7 +972,104 @@ internal void OnTextCopied(TextEventArgs e)
...
+        }
+
+        public IControl GetControlInDirection(NavigationDirection direction, IControl from)
+        {
+            return null;
+        }
+
         public IList<RoutedCommandBinding> CommandBindings { get; } = new List<RoutedCommandBinding>();
+
+        public bool IsLogicalScrollEnabled => true;
```
Comment (aelij) [inline src/AvaloniaEdit/Editing/TextArea.cs]: I think `ILogicalScrollable` members should be implemented explicitly

**Comment by aelij** [comment]
Great! Thanks :)

File: src/AvaloniaEdit/Editing/TextArea.cs [hunk 3]
```diff
@@ -1049,13 +1049,25 @@ public IControl GetControlInDirection(NavigationDirection direction, IControl fr
 
         public IList<RoutedCommandBinding> CommandBindings { get; } = new List<RoutedCommandBinding>();
 
-        public bool IsLogicalScrollEnabled => true;
+        public bool IsLogicalScrollEnabled { get; } = true;
```
Comment (aelij) [inline src/AvaloniaEdit/Editing/TextArea.cs]: Oh, not what I meant. The expression syntax is preferred, so undo these changes.

I meant that all of the `ILogicalScrollable` members should not be public. So instead of `public bool IsLogicalScrollEnabled => true`, use `bool ILogicalScrollable.IsLogicalScrollEnabled => true`

Comment (danwalmsley) [inline src/AvaloniaEdit/Editing/TextArea.cs]: Ah I see what you mean, yeah I agree that is better! :)

**Comment by aelij** [comment]
Great!
